pkg/containerrun/containerrun.go

```diff
@@ -504,28 +582,58 @@ type Stdio struct {
 // ProcessRegistry handles all the processes.
 type ProcessRegistry struct {
 	processes []Process
+	timer     *time.Timer
 	sync.Mutex
 }
 
 // NewProcessRegistry constructs a new ProcessRegistry.
 func NewProcessRegistry() *ProcessRegistry {
 	return &ProcessRegistry{
 		processes: make([]Process, 0),
+		// It should always be safe to call timer.Stop(), so it must not be nil.
+		timer: time.NewTimer(time.Millisecond),
 	}
 }
 
+// Count returns the number of processes in the registry
+func (pr *ProcessRegistry) Count() int {
+	return len(pr.processes)
+}
+
 // Register registers a process in the registry and returns how many processes are registered.
 func (pr *ProcessRegistry) Register(p Process) int {
 	pr.Lock()
 	defer pr.Unlock()
+
+	log.Debugf("Registering process %s\n", p)
 	pr.processes = append(pr.processes, p)
 	return len(pr.processes)
 }
 
+// Unregister removes a process from the registry and returns how many processes are registered.
+func (pr *ProcessRegistry) Unregister(p Process) int {
+	pr.Lock()
+	defer pr.Unlock()
+
+	log.Debugf("Unregistering process %s\n", p)
+	processes := make([]Process, 0)
```

Comment:
```suggestion
	processes := make([]Process, 0, len(pr.processes))
```
(Making it the same size in case `p` is not in `pr.processes`; since this is `len()` not capacity, this will at most be one extra).

Reply:
Can do, but doesn't really make a differences given that there will be 0, 1, or 2 processes in the array.